test/functional/rpc_net.py

```diff
@@ -333,6 +334,20 @@ def test_addpeeraddress(self):
             addrs = node.getnodeaddresses(count=0)  # getnodeaddresses re-runs the addrman checks
             assert_equal(len(addrs), 2)
 
+    def test_addrmaninfo(self):
+        self.log.info("Test addrmaninfo")
+        node = self.nodes[1]
+
+        # current ipv4 count in node's Addrman: new 1, tried 1
+        self.log.debug("Test that addrmaninfo provides correct new/tried table address count")
+        node.addpeeraddress(address="3.1.1.1", tried=True, port=8333)
```

Comment:
Maybe it would actually make sense to add the possibility of a deterministic addrman as, it e.g. we could possibly extend the existing `-addrmantest` test-only RPC. But no need to do it in this PR.

Reply:
interesting, thanks! so that's why the ci was failing. i've updated the PR to reflect this.

Reply:
I don't think so - even with the guarantee that `GetGroup()` is different, there is still a small chance that two addresses happen to hash to the same bucket in `GetNewBucket()`.

Reply:
discussed this more with @amitiuttarwar and understood that `nKey` would remain the [same](https://github.com/bitcoin/bitcoin/blob/8c4958bd4c06026dc108bc7f5f063d1f389d279b/src/addrman.cpp#L109) for a node's addrman and hence same bucket position.

i tried generating random ipv6 addresses and inserting into addrman:
```
while len(node.getnodeaddresses(count=0)) < 3:
            ipv6_address = ":".join(format(randint(0, 0xffff), '04x') for _ in range(8))
            node.addpeeraddress(address=ipv6_address, tried=False, port=8333)
```

https://github.com/bitcoin/bitcoin/pull/26988#discussion_r1134455743 would still be a concern since there's a small chance that an ipv6 address could hash to the same bucket as an ipv4 address. this would cause the assertions to check the count of different network addresses in the test to fail.

so leaving out the network coverage tests for now.

Reply:
fyi: `addpeeraddress` returns `success` indicating if inserting was successful. I'm using this to retry inserting in case there was a collision.

```
added = False
while not added:
    result = node.addpeeraddress(address="..", tried=to_tried, port=port)
    added = result["success"]
```